src/libmongoc/doc/mongoc_client_get_handshake_description.rst

```diff
@@ -0,0 +1,50 @@
+:man_page: mongoc_client_get_handshake_description
+
+mongoc_client_get_handshake_description()
+=========================================
+
+Synopsis
+--------
+
+.. code-block:: c
+
+   mongoc_server_description_t *
+   mongoc_client_get_handshake_description (mongoc_client_t *client,
+                                            uint32_t server_id,
+                                            bson_t *opts,
+                                            bson_error_t *error);
+
+Returns a description constructed from the initial handshake response to a server.
+
+Description
+-----------
+
+:symbol:`mongoc_client_get_handshake_description` is distinct from :symbol:`mongoc_client_get_server_description`. :symbol:`mongoc_client_get_server_description` returns a server description constructed from monitoring, which may differ from the server description constructed from the connection handshake.
+
+:symbol:`mongoc_client_get_handshake_description` does not attempt to establish a connection to the server if a connection was not already established. It will complete authentication on a single-threaded client if the connection has not yet been used by an operation.
+
+To establish a connection on a single-threaded client, ensure the server is selectable (the monitoring connection is the only connection made to the server). This can be done with a function like :symbol:`mongoc_client_select_server`, or any function which performs server selection.
+
+To establish a connection on a pooled client, send a command (e.g. ping) to the server.
```

Comment:
Just to confirm, `mongoc_client_get_handshake_description` is all about returning handshake data for an _application_ stream, correct? In pooled mode, this has nothing to do with monitoring streams because those don't have server IDs and are entirely internal.

Reply:
Correct, in pooled mode `mongoc_client_get_handshake_description` will not touch monitoring streams, or use a server description obtained by those monitoring streams.

In pooled mode, internally, monitoring streams they still map to server IDs. They are managed in `mongoc_server_monitor_t`, and are never used for application operations.